### PR TITLE
💨 compress the image in client and modified .gitignore

### PR DESCRIPTION
**Pull request by hongyec**
# Description 
In [issue #187](https://github.com/Shared-Reality-Lab/IMAGE-server/issues/187), we try to implement a rescaling of graphics across the board in the extension, to prevent excessively large graphics being unnecessarily sent to the server for processing. 

As the preprocess can process the image around 1200 pixel, what this pr do is to make either width or length of the image below 1200 pixel:

For examples, for 2400 \* 800 images, they will transform to 1200 \* 400 ones, and for 800 \* 2400 images, they will transform to 400 \* 1200 ones.

### CODE REVIEW

**Comment by jeffbl** [comment]
If this is ready for review, it needs to be assigned to @jaydeepsingh25 . From the title, it seems like it is only being compressed, and not resized? Can you summarize your algorithm briefly?

**Comment by hongyec** [comment]
Hi @jeffbl, just add the description for the pr, and assigned Jaydeep for review.

**Comment by jaydeepsingh25** [comment]
Also, the PR should not contain any changes under schemas/ . Please make sure you do not push these changes

**Comment by jaydeepsingh25** [comment]
Did you test it @hongyec ? , i don't think it is working. I tried with the image mentioned in https://github.com/Shared-Reality-Lab/IMAGE-server/issues/187, and i am not getting any renderings.

**Comment by jaydeepsingh25** [comment]
Also, you need to send the updated dimensions to the server, whenever you are resizing the image:
https://github.com/Shared-Reality-Lab/IMAGE-browser/blob/8282c72ef954ce2bf3360ebf24ab11a915dd538e/src/background.ts#L103

**Comment by hongyec** [comment]
@jaydeepsingh25, I just tested the starfish picture without specify the quality of the picture, and it works! Meanwhile I also change the dimensions. I will remove the change in schema as soon as possible.

**Comment by JRegimbal** [comment]
@hongyec does this absolutely need unsafe-eval? That's a massive security policy change and I don't see eval used in your PR.

**Comment by hongyec** [comment]
@jaydeepsingh25 I think it used to, because of the get pixel method, I think now I could remove them

**Comment by hongyec** [comment]
@jaydeepsingh25  removed the unsafe-eval!

**Review by jaydeepsingh25 — APPROVED** [review]
